Makefile

```diff
@@ -154,8 +155,9 @@ build/libcxx.BUILT: build/llvm.BUILT build/compiler-rt.BUILT build/wasi-libc.BUI
 	# Do the build.
 	mkdir -p build/libcxx
 	cd build/libcxx && cmake -G Ninja $(LIBCXX_CMAKE_FLAGS) \
-	    -DCMAKE_C_FLAGS="$(DEBUG_PREFIX_MAP) --sysroot=$(BUILD_PREFIX)/share/wasi-sysroot" \
-	    -DCMAKE_CXX_FLAGS="$(DEBUG_PREFIX_MAP) --sysroot=$(BUILD_PREFIX)/share/wasi-sysroot" \
+	    -DCMAKE_SYSROOT=$(BUILD_PREFIX)/share/wasi-sysroot \
+	    -DCMAKE_C_FLAGS="$(DEBUG_PREFIX_MAP) \
+	    -DCMAKE_CXX_FLAGS="$(DEBUG_PREFIX_MAP) \
```

Comment:
Why these changes?

Reply:
This is the reliable way to pass sysroot to the compiler by cmake.
It's can  handling msys2 and msys2/mingw properly

Reply:
I split it out into other PR